Code/app/src/main/java/com/cmpt362/fitcheck/firebase/Firebase.kt

```diff
@@ -0,0 +1,52 @@
+package com.cmpt362.fitcheck.firebase
+
+import com.google.android.gms.tasks.Task
+import com.google.firebase.auth.AuthResult
+import com.google.firebase.auth.FirebaseAuth
+import com.google.firebase.auth.ktx.auth
+import com.google.firebase.database.DatabaseReference
+import com.google.firebase.database.FirebaseDatabase
+import com.google.firebase.database.ktx.database
+import com.google.firebase.ktx.Firebase
+
+/**
+ * Singleton class for handling all Firebase related operations
+ */
+object Firebase {
+
+    private const val USERS_REFERENCE_NAME = "users"
+    private const val FRIENDSHIPS_REFERENCE_NAME = "friendships"
+    private const val USER_PHOTOS_REFERENCE_NAME = "user_photos"
+    private const val PHOTOS_TAGS_REFERENCE_NAME = "photos_tags"
+
+    private val auth: FirebaseAuth = Firebase.auth
+    private val database: FirebaseDatabase = Firebase.database
+
+    private val usersReference: DatabaseReference
+    private val friendshipsReference: DatabaseReference
+    private val userPhotosReference: DatabaseReference
+    private val photosTagsReference: DatabaseReference
+
+    init {
+        usersReference = database.getReference(USERS_REFERENCE_NAME)
+        friendshipsReference = database.getReference(FRIENDSHIPS_REFERENCE_NAME)
+        userPhotosReference = database.getReference(USER_PHOTOS_REFERENCE_NAME)
+        photosTagsReference = database.getReference(PHOTOS_TAGS_REFERENCE_NAME)
+    }
```

Comment:
Also are we sure that each time we call "Firebase." it's using the same original FirebaseAuth/FirebaseDatabase or are they creating new ones each time?